### DISCUSSION TITLE
`Youtube Search`

### DISCUSSION THREAD

**Discussion by OmiChrome** [tomasklaen/uosc#782] [Ideas]
#### The IDEA 💡 
searching youtube video with-in mpv !
_____________________________________________________________________

I previously used a script  `youtube-search.lua`  which I forgot where I found it, 
with the keybind `CTRL+SHIFT+S ` but now after the update it someone not working for me.
there is one [mpv-youtube-search](https://github.com/rozari0/mpv-youtube-search) but it require [mpv-zenity-open-files](https://github.com/alifarazz/mpv-zenity-open-files) which quite not work well too

so integrating YOUTUBE-SEARCH with-in UOSC input is great !

**Comment by christoph-heinrich** [comment]
https://github.com/CogentRedTester/mpv-scripts/blob/master/youtube-search.lua

**Reply by OmiChrome** [reply]
sorry, i just can't get it working
the menu which is suppose to appear asking enter the search-term or url is not appearing
by pressing [CTRL+y](https://github.com/CogentRedTester/mpv-scripts/blob/651c660aa5d9581b766ce38d9eef46a27a8b4432/youtube-search.lua#L337) i tried to change values of keybinds yet still can't :'(

**Reply by christoph-heinrich** [reply]
I can't help you with that as I don't use that script myself, I just thought that must be the one you're looking for :)

**Reply by xfzv** [reply]
@NormTurtle Not using `youtube-search.lua` myself either, but quoting the block comment at the very top of the file:

```lua
--[[ This script requires that my other scripts `scroll-list` and `user-input` be installed.
    scroll-list.lua and user-input-module.lua must be in the ~~/script-modules/ directory,
    while user-input.lua should be loaded by mpv normally.

    https://github.com/CogentRedTester/mpv-scroll-list
    https://github.com/CogentRedTester/mpv-user-input

    This script also requires a youtube API key to be entered.
    The API key must be passed to the `API_key` script-opt.
    A personal API key is free and can be created from:
    https://console.developers.google.com/apis/api/youtube.googleapis.com/

    The script also requires that curl be in the system path.
]]--
```

Have you set up all these requirements?

I guess you should rather open a discussion or an issue directly on the upstream repo if you need help about this plug-in, since this doesn't have anything to do with `uosc`.

**Reply by OmiChrome** [reply]
yes i have tried these methods , but the `scroll-list` and `user-input` script block input into the `UOSC` !
thats why i thought it would be better if it is integrated into the UOSC it self, such as it already have better tiling, input, and view-modes
:)

**Reply by uugaabuugaa** [reply]
I'm also looking for a youtube-search script that works with uosc. After migrating from mpv.net to uosc a couple of days ago, everything youtube related like youtube-upnext, quality-menu work really smooth with uosc including control buttons. On the flip side I can't find any youtube-search script that works with uosc and it's triggering my mild OCD.

> https://github.com/CogentRedTester/mpv-scripts/blob/master/youtube-search.lua

This is the only script that works for me as intended but I can't find a way to even put it on the uosc menu list. Things like script-binding or script-message-to which worked with most scripts so far isn't working either. Pretty frustrating to be honest.

>  but the scroll-list and user-input script block input into the UOSC !

No it doesn't. I'm using it with uosc. You just can't change the default keybinding or have any uosc's visual sleekness with it.

Edit: My dumbass somehow just figured out `mp.add_key_binding("x", "y", z)` in a lua script means `x script-binding y` in input.conf . Got the control buttons and submenu to work. Still hoping the uosc devs will make a better gui integration for some sort of youtube search.

**Reply by christoph-heinrich** [reply]
> No it doesn't. I'm using it with uosc.

There have been some changes to how we handle user input since that comment was made, maybe that made a difference here.

In theory it shouldn't be hard to make that youtube-search script work with uosc and make scroll-list and user-input optional, but I don't know if anyone here cares enough about that to make that happen.

If you want to give it a shot, you can use https://github.com/po5/memo as a reference on how menu + search can be used.
You'll probably want to use `search_style='palette'` and `search_debounce='submit'`.

**Reply by uugaabuugaa** [reply]
> If you want to give it a shot, you can use https://github.com/po5/memo as a reference

I just started using it instead of SimpleHistory since yesterday because of uosc-fication. But I have to admit I don't think I have nearly enough capabilities to make youtube-search look like it. Everything I have done with mpv until now is mostly editing based on some sort of template.

**Reply by itsmeipg** [reply]
@uugaabuugaa @NormTurtle I converted CogentRedTester's script for uosc: https://github.com/itsmeipg/mpv-config/blob/main/portable_config/scripts/uosc_youtube_search.lua

**Reply by uugaabuugaa** [reply]
> @uugaabuugaa @NormTurtle I converted CogentRedTester's script for uosc: https://github.com/itsmeipg/mpv-config/blob/main/portable_config/scripts/uosc_youtube_search.lua

I just took a look your script. Seems to be working pretty well. Thanks!
